### DISCUSSION TITLE
Add withTRPC on existing app using react-query

### DISCUSSION THREAD

**Discussion by testerez** [trpc/trpc#1594] [Q&A]
Hello,
I just added `trpc` to an exiting react app that was already using `react-query`. Unfortunately, as `withTRPC` is actually setting a `QueryClientProvider`, it conflicts with the one already set for the app.
Is there a way to configure `withTRPC` provider so it's using its own `queryClient`? Or any other solution that would allow multiple `react-query` configurations?

**Comment by KATT** [comment]
Hey! You'll need to remove your current `<QueryClientProvider />` if you wanna use the `withTRPC`-HOC. You can configure the query client like this: https://github.com/trpc/trpc/blob/0b86d95478f82308e96c1d62d7f2d70a94023fea/examples/next-prisma-starter/src/pages/_app.tsx#L74

**Reply by testerez** [reply]
Thanks @KATT but then how can I provide the `react-query` client to the rest of my app that uses `react-query` without trpc?

**Reply by KATT** [reply]
That should remain unchanged, you can import useQuery from react-query etc
